AppCenterData/AppCenterDataTests/MSDataTests.m

```diff
@@ -2152,4 +2159,87 @@ - (NSHTTPURLResponse *)generateResponseWithStatusCode:(NSInteger)statusCode {
+                                                                                       documentType:[MSDictionaryDocument class]
+                                                                                          partition:kMSPartitionTest documentId:documentId
+                                                                                    fromDeviceCache:NO]);
+  OCMStub([localStorageMock readWithToken:tokenResult documentId:OCMOCK_ANY documentType:OCMOCK_ANY]).andReturn(localStorageDocument);
+  
+  // Mock CosmosDB requests.
+  OCMStub([self.cosmosDbMock performCosmosDbAsyncOperationWithHttpClient:OCMOCK_ANY
+                                                             tokenResult:tokenResult
+                                                              documentId:documentId
+                                                              httpMethod:kMSHttpMethodPost
+                                                                document:OCMOCK_ANY
+                                                       additionalHeaders:OCMOCK_ANY
+                                                       additionalUrlPath:OCMOCK_ANY
+                                                       completionHandler:OCMOCK_ANY])
+  .andDo(^(NSInvocation *invocation) {
+    MSHttpRequestCompletionHandler cosmosdbOperationCallback;
+    [invocation getArgument:&cosmosdbOperationCallback atIndex:9];
+    cosmosdbOperationCallback(jsonFixture, [self generateResponseWithStatusCode:MSHTTPCodesNo200OK], nil);
+  });
+  
+  // Mock pending operation.
+  NSError *error;
+  NSDictionary *document = [NSJSONSerialization JSONObjectWithData:jsonFixture options:0 error:&error];
+  MSPendingOperation *mockPendingOperation =
+  [[MSPendingOperation alloc] initWithOperation:kMSPendingOperationCreate
+                                      partition:kMSPartitionTest
+                                     documentId:documentId
+                                       document:document
+                                           etag:@"1234"
+                                 expirationTime:[[NSDate dateWithTimeIntervalSinceNow:1000000] timeIntervalSince1970]];
+  OCMStub([localStorageMock pendingOperationsWithToken:OCMOCK_ANY]).andReturn(@[ mockPendingOperation ]);
+  
+  // Simulate being online.
+  //  currentNetworkStatus = ReachableViaWiFi;
+  //  [[NSNotificationCenter defaultCenter] postNotificationName:kMSReachabilityChangedNotification
+  //                                                      object:self.reachabilityMock];
+  
+  [self.sut processPendingOperations];
+  
+  // Then
+  id<MSRemoteOperationDelegate> strongDelegate = [MSData sharedInstance].remoteOperationDelegate;
+  XCTAssertNotNil(strongDelegate);
+  XCTAssertEqual(strongDelegate, delegateMock);
+  
+  OCMVerify([delegateMock data:self.sut didCompletePendingOperation:kMSPendingOperationCreate forDocument:localStorageDocument withError:nil]);
```

Comment:
I am having a timing issue with this test. The verification step checks even before the pending operations get complete. It goes ahead and fails saying the delegate was never invoked. Any suggestions on how I can fix this

Reply:
1. You need to set auth token at the beginning of the test. delegate isn't being called.
2. As you mentioned there is a timing issue. You need to use `XCTestExpectation` and wait until delegate is called before the verification.